osgp/platform/osgp-domain-distributionautomation/src/main/java/org/opensmartgridplatform/domain/da/measurements/elements/BitmaskMeasurementElement.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ */
+package org.opensmartgridplatform.domain.da.measurements.elements;
+
+import java.util.EnumSet;
+import java.util.Objects;
+
+import org.opensmartgridplatform.domain.da.measurements.MeasurementElement;
+
+public class BitmaskMeasurementElement extends MeasurementElement<Byte> {
+
+    public BitmaskMeasurementElement(final Byte value) {
+        super(value);
+    }
+
+    public boolean getFlag(final short position) {
+        return this.getFlag(position, true);
+    }
+
+    public boolean getFlag(final short position, final boolean zeroBased) {
+        final int index = zeroBased ? position : position - 1;
+        return (this.value & 1 << index) > 0;
+    }
+
+    public EnumSet<BitmaskFlag> asEnumSet() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The return type of this method should be an interface such as "Set" rather than the implementation "EnumSet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1319)